internal/examples/supervisor/supervisor/commander/commander.go

```diff
@@ -116,6 +128,10 @@ func (c *Commander) IsRunning() bool {
 // and if the process does not finish kills it forcedly by sending SIGKILL.
 // Returns after the process is terminated.
 func (c *Commander) Stop(ctx context.Context) error {
+	c.isStoppingMutex.Lock()
+	c.isStoppingFlag = true
```

Comment:
We never set `isStoppingFlag` to false after this. It means `Restart()` (or `Start()` after `Stop()`) will not work correctly.

Reply:
We don't have a test that verifies the scenario I describe, but it can happen in reality when the config is changed and the agent is restarted, see https://github.com/open-telemetry/opamp-go/blob/71fdce8d4b62fd3d2c3cbbc50f9765bc73a0fe15/internal/examples/supervisor/supervisor/supervisor.go#L494

Reply:
It may be worth testing the Supervisor behavior manually after this fix. The restarts happen when the Server sends a new config to the Supervisors.

Reply:
Good catch, I end up removing the isStoppingFlag and rely on the mutex lock. https://github.com/open-telemetry/opamp-go/pull/352/commits/f71ff69b8806e1d9fde956da1d8a2d4607e1f188

This will prevent a race condition between the Start/Stop (manually tested with restarts), although I think it should be handled by the Supervisor and not the commander. The current Supervisor code can be shutdown, but not stopping the supervised processes. We will need to add a sync mechanism between the `runAgentProcess` routine and the Shutdown method.